### ISSUE TITLE
Rule does not work after rebootw

### ISSUE THREAD

**Issue by sshark** [microsoft/WSL#6731]
I was following the advice from https://github.com/microsoft/WSL/issues/4585,

```
New-NetFirewallRule -DisplayName "WSL" -Direction Inbound  -InterfaceAlias "vEthernet (WSL)"  -Action Allow
```

to set up an interface to allow my browsers to connect locally to the applications in the WSL2 via HTTP. It was working until I reboot my Windows 10 20H2 [10.0.19042.867]. Next, I have to recreate the firewall interface again and it worked until the next machine reboot. Am I missing something from the command? Thanks

**Comment by AttilaSATAN** [comment]
I've just updated my win10 to W10.21H1. But still, the problem continues.
The WSL rule that I created can be seen in inbound rules and it's enabled but can't connect to the host PC. After reading @Stef-Alb 's explanation, I've disabled and reenabled the rule and it started to work.

**Comment by zejal** [comment]
Is there any real solution to this problem ? I mean without workarounds. It's annoying to have to enable again the firewall rules at each reboot (which seems frequent with Windows 11 but that's another story). It's rather unacceptable that an issue opened nearly 1 year ago is still opened !

**Comment by brumlemann** [comment]
Still a problem in W11 with WSL 2.1.5.0.

**Comment by YanekC** [comment]
The issue is still present in WSL version 2.3.26.0 .
Is there any workaround to set the GUID of the created WSL interface ?

**Comment by YanekC** [comment]
> https://learn.microsoft.com/en-us/windows/wsl/networking#mirrored-mode-networking https://learn.microsoft.com/en-us/windows/security/operating-system-security/network-security/windows-firewall/hyper-v-firewall#firewall-rules

I've tried adding the same firewall rule but using hyper-v firewall.
`New-NetFirewallHyperVRule -DisplayName "WSL Inbound" -Direction Inbound -VMCreatorId '{40E0AC32-46A5-438A-A0B2-2B479E8F2E90}' -Action Allow`

But it does not seems to work. 
It still work when I'm adding `New-NetFirewallRule -DisplayName "WSL" -Direction Inbound  -InterfaceAlias "vEthernet (WSL (Hyper-V firewall))"  -Action Allow` after each reboot. 

What am I missing ? 


I also tried mirrored mode but I had new problems using it...  
For the record I'm using WSL with Intellij IDEA
